### PR TITLE
Fix code for dataset and dataset version

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

**Comment by coryverta** [comment]
@ad-47 Please provide a PR description of this change and link to JIRA ticket(s) if they exist.

**Comment by anandjakhaniya** [comment]
@coryverta this is a bugs fixes PR about visibility for the dataset and dataset version. there is no any ticket created for this.
Description:
- fix fetching workspace datasetId using new workspace fetch logic and remove old workspace, workspaceType check from database query
- Fix `findDatasets` Dao implementation with new workspace logic

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 1]
```diff
@@ -1312,88 +1310,69 @@ public DatasetPaginationDTO findDatasets(
...
+                ModelDBServiceResourceTypes.REPOSITORY);
+      }
+
+      if (accessibleDatasetIds.isEmpty() && roleService.IsImplemented()) {
+        LOGGER.debug("Accessible Dataset Ids not found, size 0");
+        DatasetPaginationDTO emptyPaginationDTO = new DatasetPaginationDTO();
+        emptyPaginationDTO.setDatasets(Collections.emptyList());
+        emptyPaginationDTO.setRepositories(Collections.emptyList());
+        emptyPaginationDTO.setTotalRecords(0L);
+        return emptyPaginationDTO;
```
Comment (coryverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: This code is duplicated a few times, please extract to a private utility method.

**Review by coryverta — APPROVED** [review]
One small refactor, then LGTM

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java [hunk 2]
```diff
@@ -1312,88 +1310,69 @@ public DatasetPaginationDTO findDatasets(
...
-                builder.equal(
-                    repositoryRoot.get(ModelDBConstants.WORKSPACE_TYPE),
-                    workspacePredicates.get(1).getValue().getNumberValue());
-            Predicate privatePredicate =
-                builder.and(privateWorkspacePredicate, privateWorkspaceTypePredicate);
-
-            finalPredicatesList.add(privatePredicate);
+        for (GetResourcesResponseItem item : accessibleAllWorkspaceItems) {
+          if (orgWorkspaceIds.contains(String.valueOf(item.getWorkspaceId()))) {
+            accessibleDatasetIds.remove(item.getResourceId());
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:1345]: I remember having a note on this filtering being redundant, right?

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:1345]: Yes i remember that comments but here i have to filter out org_id from the my accessible ids with collaborators entities. there is no any endpoint which returns me just own entities ids and shared entity ids excluding org_id. that's why i have added these filter. if i am getting wrong on this let me know.

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java]: Extracted

File: backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java (hunk 2 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:1345]: It's fine, let's just tag a todo

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java:1345]: Added TODO

**Pull request merged**